libember_slim/Source/api.h

```diff
@@ -0,0 +1,54 @@
+/*
+   libember_slim -- ANSI C implementation of the Ember+ Protocol
+   Copyright (C) 2012-2014  L-S-B Broadcast Technologies GmbH
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#ifndef __LIBEMBER_API_H
+#define __LIBEMBER_API_H
+
+/*
+ * Preprocessor definitions for generic symbol import/export support
+ * Adapted from http://gcc.gnu.org/wiki/Visibility
+ */
+
+#if defined _WIN32 || defined __CYGWIN__
+#  define LIBEMBER_HELPER_DLL_IMPORT __declspec(dllimport)
+#  define LIBEMBER_HELPER_DLL_EXPORT __declspec(dllexport)
+#  define LIBEMBER_HELPER_DLL_LOCAL
+#else
+#  define LIBEMBER_HELPER_DLL_IMPORT __attribute__ ((visibility ("default")))
+#  define LIBEMBER_HELPER_DLL_EXPORT __attribute__ ((visibility ("default")))
+#  define LIBEMBER_HELPER_DLL_LOCAL  __attribute__ ((visibility ("hidden")))
+#endif
+
+#if defined LIBEMBER_DLL                // Define this if libember is compiled or used as a DLL/shared object
+#  ifdef LIBEMBER_HEADER_ONLY
```

Comment:
This case can be removed, because the C library can not be used as a header only library.